src/lib/media-import/status.js

```diff
@@ -0,0 +1,238 @@
+		case 'ABORTING':
+			return chalk.yellow( '⚠️' );
+		default:
+			return '';
+	}
+}
+
+function buildErrorMessage( importFailed ) {
+	let message = chalk.red( `Error: ${ importFailed.error }` );
+
+	if ( 'FAILED' === importFailed.status ) {
+		message += `
+An error occurred: ${ importFailed.filesProcessed }/${ importFailed.filesTotal } files imported.
+
+If this error persists, please contact support.
+`;
+	}
+	return message;
+}
+
+export async function mediaImportCheckStatus( {
+	app,
+	env,
+	progressTracker,
+}: MediaImportCheckStatusInput ) {
+	// Stop printing so we can pass our callback
+	progressTracker.stopPrinting();
+
+	// NO `console.log` in this function (until results are final)! It will break the progress printing.
+	const api = await API();
+
+	if ( ! currentUserCanImportForApp( app ) ) {
+		throw new Error(
+			'The currently authenticated account does not have permission to view Media import status.'
+		);
+	}
+	let overallStatus = 'Checking...';
+
+	const setProgressTrackerSuffix = () => {
+		const sprite = getGlyphForStatus( overallStatus, progressTracker.runningSprite );
+
+		const exitPrompt = '(Press ^C to hide progress. The import will continue in the background.)';
+
+		let statusMessage;
+		switch ( overallStatus ) {
+			case 'COMPLETED':
+				statusMessage = `COMPLETED ${ sprite } : The Imported files should be visible on your site ${ env.primaryDomain.name }`;
+				break;
+			case 'COMPLETING':
+				statusMessage = `COMPLETING ${ sprite } `;
+				break;
+			// Intentionally no break to get default case:
+			default:
+				statusMessage = `${ capitalize( overallStatus ) } ${ sprite }`;
+		}
+
+		const maybeExitPrompt = `${ overallStatus !== 'COMPLETED' ? exitPrompt : '' }`;
+
+		const suffix = `
+=============================================================
+Status: ${ statusMessage }
+App: ${ app.name } (${ formatEnvironment( env.type ) })
+=============================================================
+${ maybeExitPrompt }
+`;
+		progressTracker.suffix = suffix;
+	};
+
+	const setSuffixAndPrint = () => {
+		setProgressTrackerSuffix();
+		progressTracker.print();
+	};
+
+	progressTracker.startPrinting( setSuffixAndPrint );
+
+	const getResults = () =>
+		new Promise( ( resolve, reject ) => {
+			const checkStatus = async () => {
+				let mediaImportStatus;
+				try {
+					mediaImportStatus = await getStatus( api, app.id, env.id );
+				} catch ( error ) {
+					return reject( { error } );
+				}
+
+				const {
+					status,
+				} = mediaImportStatus;
+
+				const failedMediaImport = ( 'FAILED' === status );
+
+				if ( failedMediaImport ) {
+					progressTracker.setStatus( mediaImportStatus );
+					overallStatus = 'FAILED';
+					setSuffixAndPrint();
+					/* TODO: Here we will add failure details */
+					return reject( { ...mediaImportStatus, error: 'Import failed' } );
+				}
+
+				progressTracker.setStatus( mediaImportStatus );
+
+				setSuffixAndPrint();
+
+				if ( [ 'COMPLETED', 'ABORTED' ] === status ) {
```

Comment:
This doesn't seem to be working. If the import is `COMPLETED`, the `checkStatus` loop still runs. Seems to work as expected if I changed it like so:
```suggestion
				if ( [ 'COMPLETED', 'ABORTED' ].includes( status ) ) {
```

Reply:
Thanks @hanifn , I had just noticed that and fixed that in a just-submitted commit 😓